pkg/cmd/admin/top/images.go

```diff
@@ -0,0 +1,226 @@
+// which can be validated and used for showing limits usage.
+func (o *TopImagesOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, args []string, out io.Writer) error {
+	osClient, kClient, err := f.Clients()
+	if err != nil {
+		return err
+	}
+	namespace := cmd.Flag("namespace").Value.String()
+	if len(namespace) == 0 {
+		namespace = kapi.NamespaceAll
+	}
+	o.out = out
+
+	allImages, err := osClient.Images().List(kapi.ListOptions{})
+	if err != nil {
+		return err
+	}
+	o.Images = allImages
+
+	allStreams, err := osClient.ImageStreams(namespace).List(kapi.ListOptions{})
+	if err != nil {
+		return err
+	}
+	o.Streams = allStreams
+
+	allPods, err := kClient.Pods(namespace).List(kapi.ListOptions{})
+	if err != nil {
+		return err
+	}
+	o.Pods = allPods
+
+	return nil
+}
+
+// Validate ensures that a TopImagesOptions is valid and can be used to execute command.
+func (o TopImagesOptions) Validate(cmd *cobra.Command) error {
+	return nil
+}
+
+// Run contains all the necessary functionality to show current image references.
+func (o TopImagesOptions) Run() error {
+	infos := o.imagesTop()
+	Print(o.out, ImageColumns, infos)
+	return nil
+}
+
+var ImageColumns = []string{"NAME", "IMAGESTREAMTAG", "PARENTS", "USAGE", "METADATA", "STORAGE"}
+
+// imageInfo contains statistic information about Image usage.
+type imageInfo struct {
+	Image          string
+	ImageStreamTag string
+	Parents        []string
+	Usage          []string
+	Metadata       bool
+	Storage        int64
+}
+
+var _ Info = &imageInfo{}
+
+func (i imageInfo) PrintLine(out io.Writer) {
+	printValue(out, i.Image)
+	printValue(out, i.ImageStreamTag)
+	printArray(out, i.Parents)
+	printArray(out, i.Usage)
+	printBool(out, i.Metadata)
+	printSize(out, i.Storage)
+}
+
+// imagesTop generates Image information from a graph and returns this as a list
+// of imageInfo array.
+func (o TopImagesOptions) imagesTop() []Info {
+	g := graph.New()
+	addImagesToGraph(g, o.Images)
+	addImageStreamsToGraph(g, o.Streams)
+	addPodsToGraph(g, o.Pods)
+	markParentsInGraph(g)
+
+	infos := []Info{}
+	imageNodes := getImageNodes(g.Nodes())
+	for _, in := range imageNodes {
+		image := in.Image
+		istag := getImageStreamTag(g, in)
+		parents := getImageParents(g, in)
+		usage := getImageUsage(g, in)
+		metadata := len(image.DockerImageManifest) != 0 && len(image.DockerImageLayers) != 0
+		infos = append(infos, imageInfo{
+			Image:          image.Name,
+			ImageStreamTag: istag,
+			Parents:        parents,
+			Usage:          usage,
+			Metadata:       metadata,
+			Storage:        image.DockerImageMetadata.Size,
+		})
+	}
+
+	return infos
+}
+
+func getImageStreamTag(g graph.Graph, node *imagegraph.ImageNode) string {
+	for _, e := range g.InboundEdges(node, ImageStreamImageEdgeKind) {
+		streamNode, ok := e.From().(*imagegraph.ImageStreamNode)
+		if !ok {
+			continue
+		}
+		return getTag(streamNode.ImageStream, node.Image)
```

Comment:
Return all of them. Sort them by tag even generation and truncate them in the printer.